lib/tables.c

```diff
@@ -4047,6 +4772,35 @@ table_collection_dump(table_collection_t *self, const char *filename, int flags)
     int ret = 0;
     kastore_t store;
 
+    /* for setting a new UUID */
+    time_t timer;
+    size_t timestamp_size = 64;
+    char buffer[timestamp_size];
+    struct tm* tm_info;
+    size_t provenance_size = 9 + UUID4_LEN;
+    char *provenance_str;
+
+    provenance_str = malloc(provenance_size);
+    if (provenance_str == NULL) {
+        ret = MSP_ERR_NO_MEMORY;
+        goto out;
+    }
+
+    time(&timer);
+    tm_info = localtime(&timer);
+    strftime(buffer, timestamp_size, "%Y-%m-%dT%H:%M:%S", tm_info);
+    snprintf(provenance_str, provenance_size, "{\"uuid\"=%s}", self->uuid);
```

Comment:
how's this?

Reply:
Looks good, but I'm not sure we need to bother with the provenance stuff right now.